src/registry.py

```diff
@@ -148,7 +148,12 @@ def close(registry, filename, private=True):
                 if value._showDefault:
                     lines.append('#\n')
                     try:
-                        x = value.__class__(value._default, value._help)
+                        # We set setDefault to False and manually call
+                        # Value._setValue, just in case the class inherits
+                        # Value.setValue to set some global state (#1349)
+                        x = value.__class__(value._default, value._help,
+                            setDefault=False)
+                        x._setValue(value._default, inherited=False)
```

Comment:
Why call `._setValue`?

Reply:
```
ERROR 2024-10-22T21:25:58 Exception printing default value of supybot.user:
Traceback (most recent call last):
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 161, in close
    lines.append('# Default value: %s\n' % x)
                 ~~~~~~~~~~~~~~~~~~~~~~~~^~~
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 684, in __str__
    s = Value.__call__(self)
        ^^^^^^^^^^^^^^^^^^^^
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 568, in __call__
    return self.value
           ^^^^^^^^^^
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 243, in __getattr__
    self.__nonExistentEntry(attr)
  File "/home/claire/.local/pipx/venvs/limnoria/lib/python3.12/site-packages/supybot/registry.py", line 225, in __nonExistentEntry
    raise NonExistentRegistryEntry(s)
supybot.registry.NonExistentRegistryEntry: 'value' is not a valid entry in 'unset'
```

Reply:
```suggestion
                        x.value = value._default
```

How about this? It works too, and probably spares some duplicate work (as `_setValue` is recursive); in addition to triggering whatever side-effect plugins might have in their `_setValue` implementations.

Reply:
Done, thank you